main.py

```diff
@@ -0,0 +1,45 @@
+
+import time
+import speech_recognition as sr
+
+r = sr.Recognizer()
+m = sr.Microphone() # Set appropriate device index, e.g `device_index=3`.
+
+# this is called from the background thread
+def callback(recognizer, audio):
+    print("Processing audio input ...")
+    # received audio data, now we'll recognize it using Google Speech Recognition
+    try:
+        # for testing purposes, we're just using the default API key
+        # to use another API key, use `r.recognize_google(audio, key="GOOGLE_SPEECH_RECOGNITION_API_KEY")`
+        # instead of `r.recognize_google(audio)`
+        speech = recognizer.recognize_google(audio)
```

Comment:
Is there any way we can set it up so that we can expect speech?

e.g. we have apps called "Tutor", "Reader" and "News". Can we expect the user to say one of those words, and get Google to give us the most likely?

If possible, this would likely reduce speech recognition errors (esp. in more unknown environments e.g. quiz answers)

Reply:
We can do a fault-tolerant substring search on the returned speech transcription. If it does not work well, we can still use Google's [speech context argument ](https://cloud.google.com/speech-to-text/docs/context-strength).